### PR TITLE
Decimal.distance(to:) was mis-implemented

### PR DESCRIPTION
**Pull request by Dante-Broggi**
Fix `Decimal.distance(to:)`
Add tests
Strip end of line whitespace

Resolves [SR-6785](https://bugs.swift.org/browse/SR-6785).

### CODE REVIEW

File: test/stdlib/TestDecimal.swift [hunk 1]
```diff
@@ -240,6 +240,7 @@ class TestDecimal : TestDecimalSuper {
                 //Strideable methods
                 expectEqual(Decimal(i).advanced(by: Decimal(j), Decimal(i+j), "\(Decimal(i).advanced(by: Decimal(j)) == \(i) + \(j)"
                 expectEqual(Decimal(i).distance(to: Decimal(j), Decimal(j-i), "\(Decimal(i).distance(to: Decimal(j)) == \(j) - \(i)")
+                expectEqual(Decimal(i).advanced(by: Decimal(i).distance(to: Decimal(j)), Decimal(j), "Decimal(i).advanced(by: Decimal(i).distance(to: Decimal(j)) == Decimal(j)")
```
Comment (xwu) [inline test/stdlib/TestDecimal.swift]: Some string interpolation is missing in the last argument there.

File: stdlib/public/SDK/Foundation/Decimal.swift [hunk 2]
```diff
@@ -24,13 +24,13 @@ extension Decimal {
 
     public static let pi = Decimal(_exponent: -38, _length: 8, _isNegative: 0, _isCompact: 1, _reserved: 0, _mantissa: (0x6623, 0x7d57, 0x16e7, 0xad0d, 0xaf52, 0x4641, 0xdfa7, 0xec58))
 
-    public var exponent: Int {
```
Comment (xwu) [inline stdlib/public/SDK/Foundation/Decimal.swift]: Please revert any changes in whitespace that aren't related to the bug being fixed.

File: test/stdlib/TestDecimal.swift [hunk 3]
```diff
@@ -237,6 +237,10 @@ class TestDecimal : TestDecimalSuper {
                 expectEqual(Decimal(i*j), Decimal(i) * Decimal(j), "\(Decimal(i*j)) == \(i) * \(j)")
                 expectEqual(Decimal(i+j), Decimal(i) + Decimal(j), "\(Decimal(i+j)) == \(i)+\(j)")
                 expectEqual(Decimal(i-j), Decimal(i) - Decimal(j), "\(Decimal(i-j)) == \(i)-\(j)")
+                //Strideable methods
```
Comment (xwu) [inline test/stdlib/TestDecimal.swift]: Nit: please follow the prevailing comment style in this file by putting a space between `//` and `Strideable`.

**Review by xwu — COMMENTED** [review]
Nice catch! Just some minor comments.

File: test/stdlib/TestDecimal.swift [hunk 4]
```diff
@@ -237,6 +237,10 @@ class TestDecimal : TestDecimalSuper {
                 expectEqual(Decimal(i*j), Decimal(i) * Decimal(j), "\(Decimal(i*j)) == \(i) * \(j)")
                 expectEqual(Decimal(i+j), Decimal(i) + Decimal(j), "\(Decimal(i+j)) == \(i)+\(j)")
                 expectEqual(Decimal(i-j), Decimal(i) - Decimal(j), "\(Decimal(i-j)) == \(i)-\(j)")
+                // Strideable methods
+                expectEqual(Decimal(i).advanced(by: Decimal(j)), Decimal(i+j), "\(Decimal(i).advanced(by: Decimal(j))) == \(i) + \(j)")
+                expectEqual(Decimal(i).distance(to: Decimal(j)), Decimal(j-i), "\(Decimal(i).distance(to: Decimal(j))) == \(j) - \(i)")
+                expectEqual(Decimal(i).advanced(by: Decimal(i)).distance(to: Decimal(j)), Decimal(j), "Decimal(i).advanced(by: Decimal(i).distance(to: Decimal(j)) == Decimal(j)")
```
Comment (xwu) [inline test/stdlib/TestDecimal.swift]: Still missing a paren within your string :)
And if you're following the style of earlier lines, consider escaping appropriate portions of each side of the equation in the string with `\()`.

Comment (huonw) [reply test/stdlib/TestDecimal.swift]: This currently looks like it's computing `j - (i + i)`, but maybe it should be `(i + j) - i`:

```swift
Decimal(i).distance(to: Decimal(i).advanced(by: Decimal(j)))
```

Or `i + (j - i)`:

```swift
Decimal(i).advanced(by: Decimal(i).distance(to: Decimal(j)))
```

**Comment by CodaFi** [comment]
This is wrong in corelibs as well. Let me generate a radar and put it on Foundation so they take a look here.

**Comment by CodaFi** [comment]
For reference rdar://57273727

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Comment by tbkka** [comment]
Closing:  This code is no longer in the Swift project.

**Pull request closed without merging**